File: 백준/Silver/9461. 파도반 수열/파도반 수열.py
t = int(input())

for _ in range(t):
    n = int(input())
    
    dp = [0] * (max(n+1, 6))
    
    dp[1] = dp[2] = dp[3] = 1
    
    for i in range(4, n+1):
        dp[i] = dp[i-2] + dp[i-3]
    
    print(dp[n])

File: 백준/Silver/9461. 파도반 수열/파도반 수열.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on("line", function (line) {
  input.push(line.trim());
});

rl.on("close", function () {
  const t = Number(input[0]);
  const arr = Array.from({ length: 100 }, () => 0);
  arr[1] = 1;
  arr[2] = 1;
  arr[3] = 1;
  for (let i = 4; i < 101; i++) {
    arr[i] = arr[i - 2] + arr[i - 3];
  }

  for (let i = 0; i < t; i++) {
    const n = Number(input[i + 1]);
    console.log(arr[n]);
  }
});
